pkg/apis/core/validation/validation.go

```diff
@@ -4351,10 +4352,24 @@ func ValidateConfigMap(cfg *core.ConfigMap) field.ErrorList {
 		for _, msg := range validation.IsConfigMapKey(key) {
 			allErrs = append(allErrs, field.Invalid(field.NewPath("data").Key(key), key, msg))
 		}
+		// check if the value stored consists entirely of valid UTF-8-encoded runes
+		if !utf8.ValidString(cfg.Data[key]) {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("data").Key(key), key, field.ErrorTypeInvalid.String()))
+		}
+		// check if we have a duplicate key in the other bag
+		if _, isValue := cfg.BinaryData[key]; isValue {
+			allErrs = append(allErrs, field.Duplicate(field.NewPath("data").Key(key), key))
```

Comment:
this returns an error that doesn't explain it's a duplicate of a key in the binaryData field... need to indicate that in the error

Reply:
Done